dll/win32/setupapi/cfgmgr.c

```diff
@@ -7489,7 +7489,7 @@ CM_Request_Device_Eject_ExA(
     TRACE("CM_Request_Device_Eject_ExA(%lx %p %s %lu %lx %p)\n",
           dnDevInst, pVetoType, debugstr_a(pszVetoName), ulNameLength, ulFlags, hMachine);
 
-    if (pszVetoName == NULL && ulNameLength == 0)
+    if (pszVetoName == NULL && ulNameLength != 0)
```

Comment:
This fix makes sense, though MS doc is a bit inconsistent.
https://learn.microsoft.com/en-us/windows/win32/api/cfgmgr32/nf-cfgmgr32-cm_request_device_eject_exw

While here, you might want to double-check if any length (0 - MAX_PATH) is accepted for the buffer.
Then add a check and/or a comment about that?

Reply:
@TAN-Gaming Serge has actually a point here: Does windows check whether ulNameLength, if non-zero, must be equal to MAX_PATH, or not ?

Also, unregarding the MAX_PATH remark, I think this test should be refined better here (in the ANSI function). Because iif `pszVetoName` is not NULL, and `ulNameLength` is equal to zero, the code continues and allocates a zero-sized buffer below, which is invalid.

I guess a better check would be:
```c
if ( (pszVetoName == NULL && ulNameLength != 0) || (pszVetoName != NULL && ulNameLength == 0) )
{
    /* fail here! */
}
```

(Note that I'm even wondering whether this modified check should also be done in the unicode version of the function...)

Reply:
Based on the test results, Windows can successfully eject the device when `pszVetoName != NULL && ulNameLength == 0` with both ANSI and Unicode version of the function.